public/app/features/alerting/unified/utils/rules.ts

```diff
@@ -533,3 +533,18 @@ export function isRecordingRuleByType(type?: RuleFormType) {
 export function isDataSourceManagedRuleByType(type?: RuleFormType) {
   return isCloudAlertingRuleByType(type) || isCloudRecordingRuleByType(type);
 }
+
+/*
+ * Grab the UID from either a rulerRule definition or a Prometheus rule definition, only Grafana-managed rules will have a UID.
+ */
+export function getRuleUID(rulerRule?: RulerRuleDTO, promRule?: Rule) {
```

Comment:
I'm not sure this function brings much value. It's almost the same as the `getRuleUid` from `rule-list/components/util.ts` but it additionally handles DMA rules which don't have `uid`
Maybe we could check first if a rule is GMA and then use the second version for GMA rules instead of having this function?